index.coffee

```diff
@@ -146,12 +147,21 @@ class ResumableRequest extends stream.Readable
 			@bytesTotal ?= parseInt(response.headers['content-length'])
 			@_reportProgress()
 			@_reportProgress.flush()
-			@response = cloneResponse(response).on('data', @push.bind(this))
+			@response = cloneResponse(response)
+			@destinations.forEach ([ dest, opts ]) =>
+				@response.pipe(dest, opts)
+			delete @destinations
```

Comment:
A delete can be pretty slow and will break the two places that use `@destinations`, is there a reason not to use `@destinations = []` instead to reset it?

Reply:
I just thought I’d save an array allocation — the variable is only ever used until the first response is received. Will update.

Reply:
For the performance side the delete causes a hidden class transition, which I believe is far more expensive than an array allocation.  It also has the potential to take the whole thing into a slow mode (whether it would for this case or not I don't know enough to say though)